src/realm/object-store/sync/app_utils.cpp

```diff
@@ -45,11 +45,69 @@ AppUtils::find_header(const std::string& key_name, const std::map<std::string, s
     return nullptr;
 }
 
-util::Optional<AppError> AppUtils::check_for_errors(const Response& response)
+std::optional<AppUtils::UrlComponents> AppUtils::split_url(std::string url)
+{
+    UrlComponents comp;
+    // Find the position of the scheme separator "://"
+    size_t scheme_end_pos = url.find("://");
+    if (scheme_end_pos == std::string::npos) {
+        // Missing scheme separator
+        return std::nullopt;
+    }
+    comp.scheme = url.substr(0, scheme_end_pos);
+    url.erase(0, scheme_end_pos + std::char_traits<char>::length("://"));
+
+    // Find the first slash "/"
+    size_t host_end_pos = url.find("/");
+    // No server provided
+    if (url.empty() || host_end_pos == 0) {
+        return std::nullopt;
+    }
+    if (host_end_pos == std::string::npos) {
+        // No path/query/components section
+        comp.server = url;
+        return std::move(comp);
+    }
+    comp.server = url.substr(0, host_end_pos);
+    comp.request = url.substr(host_end_pos);
+    return std::move(comp);
+}
+
+bool AppUtils::is_success_status_code(int status_code)
+{
+    return status_code == 0 || (status_code < 300 && status_code >= 200);
```

Comment:
They are used when a fake Response object is generated to provide an error - this will actually go away with the changes from #7241 to use a new StatusWith<>/AppErrorWith<> object instead of the fake error responses.